src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,157 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        Node(T value, Node<T> prev, Node<T> next) {
+            this.value = value;
+            this.prev = prev;
+            this.next = next;
+        }
+    }
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> nodeToAdd = new Node<T>(value, tail, null);
+        if (tail == null) {
+            head = nodeToAdd;
+        } else {
+            tail.next = nodeToAdd;
+        }
```

Comment:
Be careful with linking Nodes. When adding the first node (when tail is null), you should also consider initializing 'head' with the new node if it is the first element being added to the list.